subscribe.go

```diff
@@ -0,0 +1,80 @@
+package bps
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+)
+
+var (
+	subReg   = map[string]SubscriberFactory{}
+	subRegMu sync.Mutex
+)
+
+// ----------------------------------------------------------------------------
+
+// SubMessage defines a subscription message details.
+type SubMessage interface {
+	// Data returns raw (serialized) message data.
+	Data() []byte
+}
+
+// Handler defines a message handler.
+type Handler interface {
+	Handle(context.Context, SubMessage) error
+}
+
+// HandlerFunc is a func-based handler adapter.
+type HandlerFunc func(context.Context, SubMessage) error
+
+// Handle handles a single message.
+func (f HandlerFunc) Handle(ctx context.Context, msg SubMessage) error {
+	return f(ctx, msg)
+}
+
+// ----------------------------------------------------------------------------
+
+// Subscriber defines the main subscriber interface.
+type Subscriber interface {
+	// Subscribe subscribes for topic messages and blocks till context is cancelled or error occurs.
```

Comment:
```
... blocks until an error occurs.
```

one more consideration: handlers need to be able to exit gracefully, should we add an `ErrStop` or should we let the user deal with it:

```go
var MyOwnErr = errors.New(..)

err := client.Subscribe("topic", bps.HanderFunc(func(msg SubMessage) error {
  return MyOwnErr
})
if err == MyOwnErr {
   return nil // ugly, I think, better add ErrStop?
}
```

Reply:
I'll do `bps.Done` for it - no `Err` prefix at all, just `bps.Done` (to be really nicely readable).